ui/packages/catalog-ui-search/src/main/webapp/js/controllers/cesium.layerOrdering.spec.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import { expect } from 'chai'
+import { CesiumLayerOrdering } from './cesium.layerCollection.controller'
+
+describe('CesiumLayerOrdering', () => {
+  const checkOrdering = ({ actual, expected }) =>
+    expect(actual).to.have.same.ordered.members(expected)
+
+  describe('addLayer()', () => {
+    it('Can add layer to empty ordering', () => {
+      const newLayerOrder = CesiumLayerOrdering.addLayer({
+        prev: [],
+        cur: ['a', 'b', 'c', 'd', 'e'],
+        layer: 'a',
+      })
+      checkOrdering({ actual: newLayerOrder, expected: ['a'] })
+    })
+    it('Adds layer to correct location', () => {
+      const prev = ['b', 'd']
+      const cur = ['a', 'b', 'c', 'd', 'e']
+      const testData = [
+        { layer: 'a', expectedOrder: ['a', 'b', 'd'] },
+        { layer: 'c', expectedOrder: ['b', 'c', 'd'] },
+        { layer: 'e', expectedOrder: ['b', 'd', 'e'] },
```

Comment:
❓ Could we break this into three separate tests? Either at the same testing level with separate `it()` declarations ... or, if `mocha` supports it, nesting the `it()` declarations in your loop. The names could be as follows: 
```
Adds layer to correct location at the beginning
Adds layer to correct location in the middle
Adds layer to correct location at the end
```

Reply:
Added parameterized tests